lib/src/qr_code.dart

```diff
@@ -214,38 +214,30 @@ List<int> _createBytes(QrBitBuffer buffer, List<QrRsBlock> rsBlocks) {
   return data;
 }
 
-int _lengthInBits(int mode, int type) {
-  if (1 <= type && type < 10) {
-    // 1 - 9
-    return switch (mode) {
-      qr_mode.modeNumber => 10,
-      qr_mode.modeAlphaNum => 9,
-      qr_mode.mode8bitByte => 8,
-      qr_mode.modeKanji => 8,
-      _ => throw ArgumentError('mode:$mode')
+int _lengthInBits(int mode, int type) => switch (type) {
```

Comment:
Watch this talk - https://www.youtube.com/watch?v=dEyPP7a46TQ

and see if you can do one step better w/ records.